src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -907,6 +962,11 @@ internal void InitializePropertyCache()
                 {
                     if (property.IsExtensionData)
                     {
+                        if (EffectiveUnmappedMemberHandling is JsonUnmappedMemberHandling.Disallow)
+                        {
+                            ThrowHelper.ThrowInvalidOperationException_ExtensionDataConflictsWithUnmappedMemberHandling(Type, property);
+                        }
```

Comment:
This predates this change but I find it odd that this logic is duplicated. Could there be further abstraction with `CacheMember`?

Reply:
I had the exact same thought when working on this. We should consolidate the logic, but it takes a bit of infrastructural refactoring beyond the narrow scope of this feature. I figured I'd leave as-is for now.